vulcanai2/tests/models/test_layers.py

```diff
@@ -1,42 +1,196 @@
+        assert (torch.equal(starting_bias, new_bias) is False)
+        assert (round(new_bias.std().item(), 1) == 0.0)
+        assert (int(new_bias.mean().item()) == 0)
+
+    def test_conv_selu_bias_change(self, conv_unit):
+        """Confirm SELU bias init properties hold for conv net."""
+        starting_bias = copy.deepcopy(conv_unit._kernel.bias)
+
+        conv_unit.bias_init = selu_bias_init_
+        conv_unit._init_bias()
+        new_bias = copy.deepcopy(conv_unit._kernel.bias)
+        assert (torch.equal(starting_bias, new_bias) is False)
+        assert (round(new_bias.std().item(), 1) == 0.0)
+        assert (int(new_bias.mean().item()) == 0)
+
+
+class TestSeluInitTrain:
+    """To test selu initialization properties hold during training."""
+
+    @pytest.fixture
+    def dnn_class(self):
+        """Create DenseNet with no prediction layer."""
+        return DenseNet(
+            name='Test_DenseNet_class',
+            in_dim=(200),
+            activation=torch.nn.SELU(),
+            num_classes=10,
+            config={
+                'dense_units': [100],
+                'dropout': [0.3],
+            },
+            optim_spec={'name': 'Adam', 'lr': 0.001}
+        )
+
+    @pytest.fixture
+    def cnn_class(self):
+        """Create ConvNet with prediction layer."""
+        return ConvNet(
+            name='Test_ConvNet_class',
+            in_dim=(1, 28, 28),
+            activation=torch.nn.SELU(),
+            config={
+                'conv_units': [
+                    {
+                        "in_channels": 1,
+                        "out_channels": 16,
+                        "kernel_size": (5, 5),
+                        "stride": 2
+                    },
+                    {
+                        "in_channels": 16,
+                        "out_channels": 1,
+                        "kernel_size": (5, 5),
+                        "stride": 1,
+                        "padding": 2
+                    }]
+            },
+            num_classes=10
+        )
+
+    def test_selu_trained_dense(self, dnn_class):
+        """Confirm SELU weight and bias properties hold for a dense net."""
+        std = round(math.sqrt(1. / 200), 1)
+        test_input = torch.ones([10, *dnn_class.in_dim]).float()
+        test_target = torch.LongTensor([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+        test_dataloader = DataLoader(TensorDataset(test_input, test_target))
+        dnn_class.fit(test_dataloader, test_dataloader, 5)
+        trained = copy.deepcopy(dnn_class.network.dense_0._kernel)
+        assert (round(trained.weight.std().item(), 1) == std)
+        assert (int(trained.weight.mean().item()) == 0.0)
+        assert (round(trained.bias.std().item(), 1) == 0.0)
+        assert (int(trained.bias.mean().item()) == 0.0)
+
+    def test_selu_trained_conv(self, cnn_class):
+        """Confirm SELU weight and bias properties hold for a conv net."""
+        std = round(math.sqrt(1. / 25), 1)
```

Comment:
how did you calculate the fan_in for conv?

Reply:
@rfratila I was doing it manually when setting up the tests. I can set it up to do it automatically if you'd like?

Reply:
if possible having it done automatically would be preferred in case the fixture itself changes. thanks!

Reply:
@rfratila Just pushed to automatically calculate

Reply:
Thanks @joemehltretter ! Just made one last comment but it should be good to go afterwards :)